### PR TITLE
Initial fixed-size data structures impl

### PR DESCRIPTION
**Pull request by fckt**
- Implement from hex string/to hex string format

### CODE REVIEW

File: src/hash.rs [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+			pub fn low_u64(&self) -> u64 {
+				let mut ret = 0u64;
+				for i in 0..min($size, 8) {
+					ret |= (self.0[$size - 1 - i] as u64) << (i * 8);
+				}
+				ret
+			}
+		}
+
+		// impl str::FromStr for $from {
```
Comment (NikVolf) [inline src/hash.rs]: Remove or uncomment please )

Comment (fckt) [reply src/hash.rs]: I'm going to implement in today

File: src/hash.rs [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+    #[doc(hidden)]
+    __variant2,
+}
+
+extern {
+    fn memcmp(cx: *const c_void, ct: *const c_void, n: usize) -> i32;
+}
+
+/// Get the SHA3 (i.e. Keccak) hash of the empty bytes string.
+pub const SHA3_EMPTY: H256 = H256( [0xc5, 0xd2, 0x46, 0x01, 0x86, 0xf7, 0x23, 0x3c, 0x92, 0x7e, 0x7d, 0xb2, 0xdc, 0xc7, 0x03, 0xc0, 0xe5, 0x00, 0xb6, 0x53, 0xca, 0x82, 0x27, 0x3b, 0x7b, 0xfa, 0xd8, 0x04, 0x5d, 0x85, 0xa4, 0x70] );
```
Comment (NikVolf) [inline src/hash.rs]: No need in this PR for this

File: src/hash.rs [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+
+/// The SHA3 of the RLP encoding of empty data.
+pub const SHA3_NULL_RLP: H256 = H256( [0x56, 0xe8, 0x1f, 0x17, 0x1b, 0xcc, 0x55, 0xa6, 0xff, 0x83, 0x45, 0xe6, 0x92, 0xc0, 0xf8, 0x6e, 0x5b, 0x48, 0xe0, 0x1b, 0x99, 0x6c, 0xad, 0xc0, 0x01, 0x62, 0x2f, 0xb5, 0xe3, 0x63, 0xb4, 0x21] );
```
Comment (NikVolf) [inline src/hash.rs]: ditto

File: src/hash.rs [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+
+/// The SHA3 of the RLP encoding of empty list.
+pub const SHA3_EMPTY_LIST_RLP: H256 = H256( [0x1d, 0xcc, 0x4d, 0xe8, 0xde, 0xc7, 0x5d, 0x7a, 0xab, 0x85, 0xb5, 0x67, 0xb6, 0xcc, 0xd4, 0x1a, 0xd3, 0x12, 0x45, 0x1b, 0x94, 0x8a, 0x74, 0x13, 0xf0, 0xa1, 0x42, 0xfd, 0x40, 0xd4, 0x93, 0x47] );
```
Comment (NikVolf) [inline src/hash.rs]: ditto

File: src/hash.rs [hunk 5]
```diff
@@ -0,0 +1,434 @@
...
+
+
+/// Return `s` without the `0x` at the beginning of it, if any.
+pub fn clean_0x(s: &str) -> &str {
```
Comment (NikVolf) [inline src/hash.rs]: this is not needed
the same the function that uses it

File: src/hash.rs [hunk 6]
```diff
@@ -0,0 +1,434 @@
...
+        impl fmt::Debug for $from {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                for i in &self.0[..] {
+                    write!(f, "{:02x}", i)?;
+                }
+                Ok(())
+            }
+        }
+
+        impl fmt::Display for $from {
```
Comment (NikVolf) [inline src/hash.rs]: Not sure if we want to use parity-specific formatting for hex here

better use the same as in Debug impl

File: src/hash.rs [hunk 7]
```diff
@@ -0,0 +1,434 @@
...
+                write!(f, "…")?;
+                for i in &self.0[$size - 2..$size] {
+                    write!(f, "{:02x}", i)?;
+                }
+                Ok(())
+            }
+        }
+
+        impl Copy for $from {}
+        #[cfg_attr(feature="dev", allow(expl_impl_clone_on_copy))]
```
Comment (NikVolf) [inline src/hash.rs]: We don't have any dev features, should be removed and included or not included without dependency on it

**Pull request merged**